bundles/docker-ubuntu1604/Dockerfile

```diff
@@ -1,17 +1,40 @@
+# This image builds the an ubuntu image with the docker version informed.
+# More info: https://docs.docker.com/engine/install/ubuntu/#install-using-the-repository
 FROM ubuntu:16.04
+
+# DOCKER_VERSION stores the value informed to via arg to build the image.
+# **Note**: You can test it out with docker build --build-arg DOCKER_VERSION=20.10.17 .
+ARG DOCKER_VERSION
```

Comment:
In order to speed up building multiple docker versions this should be as low as possible in the file to take advantage of caching

Reply:
Hi @emosbaugh,

Thank you for your time and reply. 

The changes in this PR still having the same quantity of layers. So, we are not making it big at all.
Also, we still using the same options/flags to **only** install what is required. 

We are here only:
- fixing the issue _(it is not possible to build these images on master so,  I believe that those are only used for tests purposes)_
- replacing the deprecated command 
- ensuring that we are following the official recommendation

**However, we still using the same options/flags to just install what is required. Nothing changes about that.**

If the file is big it is **only** because of the comments _(which helps us to better understand each step)_ and layout which improve the readability/maintainability.

Reply:
I changed it. Please, let me know if I got what do you mean or if I still missing something here. 
Thank you a lot for your time and review 🥇